File: diary_sample/src/main/java/com/example/diary/global/util/Response.java
package com.example.diary.global.util;

import lombok.Builder;
import lombok.Getter;
import org.springframework.http.HttpStatus;

import static org.springframework.http.HttpStatus.OK;

@Getter @Builder
public class Response<T>{
    private HttpStatus httpStatus;
    private int statusNumber;
    private T data;

    public static <T> Response<?> result(T data, HttpStatus status) {
        return Response.builder()
                .httpStatus(status)
                .statusNumber(status.value())
                .data(data)
                .build();
    }

}

File: diary/src/main/kotlin/sample/diary/global/util/Response.kt
package sample.diary.global.util

import org.springframework.http.HttpStatus

class Response<T>(
    val httpStatus: HttpStatus,
    val statusNumber: Number,
    val data: T
){
    companion object {
        fun <T> result(data: T, status: HttpStatus): Response<T> {
            return Response(status, status.value(), data)
        }
    }
}
